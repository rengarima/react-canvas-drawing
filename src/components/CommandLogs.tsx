import React, { Fragment } from "react";
import {Logs} from "../types/component";

export const CommandLogs = ({logs}:Logs): JSX.Element => {
    return logs ?
     (
        <div aria-label="command-logs">
            <ol>
                {
                    logs.map( (value, i) => {
                        return <li key={i}>{value.command} </li>
                    })
                }
            </ol>
        </div>

    ): <Fragment/>;

};
