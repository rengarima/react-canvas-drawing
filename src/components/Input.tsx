import React, {useState} from "react";
import {InputProps} from "../types/InputProps";

export const Input = ({
    id, placeholder, icon, error, type = "text",executeInput

                      }:InputProps): JSX.Element => {
    const dangerStyle = error ? " is-danger": "";

    const [value, setValue ] = useState();

    function handleChange(e: any): void {
        const inputVal = e.target.value;
        setValue(inputVal);

    }

    const handleOnKeyPress = (e:any) =>{
        e.key === "Enter" && executeInput(value);
    }

    return (
        <div className="columns">
            <div className="column is-fullhd" >
                <p className="control has-icons-left">
                    <input aria-label={id}
                        className={"input is-rounded" + dangerStyle } type={type}
                           placeholder={placeholder}
                           onKeyPress={handleOnKeyPress}
                           onChange={(value: any) => {
                                handleChange(value);
                            }}/>
                    <span className="icon is-small is-left">
                      <i className={icon}></i>
                    </span>
                </p>
                { error && <p
                    className="help is-danger"
                    style={{ fontSize: 15, overflowWrap: "normal" }}
                >
                    &nbsp;{error + ""}
                </p> }
            </div>
        </div>
    );
}
