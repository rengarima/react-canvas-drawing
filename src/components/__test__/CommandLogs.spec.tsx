import React from 'react';
import { render, fireEvent, RenderResult } from '@testing-library/react';
import {Input} from '../Input';
import {CommandLogs} from "../CommandLogs";

let documentBody: RenderResult;
describe('<CommandLogs />', () => {
    beforeEach(() => {
        const logs = [{command:"Log 1"}]
        documentBody = render(<CommandLogs
            logs={logs}
        />);
    });
    it('CommandLogs is rendered', () => {
        expect(documentBody.getByLabelText("command-logs")).toBeDefined();
        expect(documentBody.getByText("Log 1")).toBeDefined();
    });

    it("matches snapshot", () => {
        const { baseElement } = documentBody;
        expect(baseElement).toMatchSnapshot();
    });
});
