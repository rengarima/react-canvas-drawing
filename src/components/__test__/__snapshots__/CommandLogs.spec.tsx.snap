// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommandLogs /> matches snapshot 1`] = `
<body>
  <div>
    <div
      aria-label="command-logs"
    >
      <ol>
        <li>
          Log 1
           
        </li>
      </ol>
    </div>
  </div>
</body>
`;
